#28 Feature:  Implementing landing-page.

Niyonkuru1 opened the pull request:
What does this PR do?

- Create UI for landing-page

Description of Task to be completed?

- Landing Page UI is created. It includes three components; footer, navbar and LandingPage.  

- It should be responsive.

- It should perfectly resemble to the corresponding design in the mockup

-  UI skeleton for loading displayed

How should this be manually tested?
- Clone the repo
- checkout to branch ft-landing-page-28
- pull the changes by git pull
- install all packages by running npm install
- start app by running npm run start.
-  It should dispaly all contents of the landing page as depicted in the design

Any background context you want to provide?

- [ Work In Progress
 ]; 

What are the relevant pivotal tracker stories?
[Trello Board # 28](https://trello.com/c/7MYeedeR/28-landing-page)

Screenshots (if appropriate)

Xldivin committed:
chore(setup): setup docker

- setup docker
- setup react frontend image
- run the frontend image through the container

[ WIP #19]

Xldivin committed:
chore(setup):setup actions

setup github actions
runs test through github actions

[WIP #27]

Xldivin committed:
fx(setup):setup actions

added test badge

[ Finishes #27]

rachelakimana committed:
chore(configure): configure codeclimate

configure codeclimate
added badges for code analysis in a readme file

[WIP #27]

Niyonkuru1 committed:
bug(fix): fix routing in react-setup project

- adding dist folder and .gitignore to the .gitignore
- adding historyApiFallback to the devserver object in the webpack.config.js file and setting it to true, to allow other routing path other than /. This fixes the routing bug.
- adding the home component and configure it in the app so that path /home should run it
- adding also the aboutpage component to be routed by path /about

Finishes #2

Niyonkuru1 committed:
bug(fix): fix routing in react-setup project

- refactoring the src/app.js by removing extra comments

- Finishes #2

rachelakimana committed:
chore(configure): configure codeclimate and heroku

configure codeclimate and heroku
added CI/CD for codeclimate and heroku

[WIP #27]

chore(configure): configure codeclimate and heroku

configure codeclimate and heroku
added CI/CD for codeclimate and heroku

[WIP #27]

chore(configure): configure codeclimate and heroku

configure codeclimate and heroku
added CI/CD for codeclimate and heroku

[WIP #27]

chore(configure): configure codeclimate and heroku

configure codeclimate and heroku
added CI/CD for codeclimate and heroku

[WIP #27]

chore(configure): configure codeclimate and heroku

configure codeclimate and heroku
added CI/CD for codeclimate and heroku

[WIP #27]

chore(configure): configure codeclimate and heroku

configure codeclimate and heroku
added CI/CD for codeclimate and heroku

[WIP #27]

chore(configure): configure codeclimate and heroku

configure codeclimate and heroku
added CI/CD for codeclimate and heroku

[WIP #27]

chore(configure): configure codeclimate and heroku

configure codeclimate and heroku
added CI/CD for codeclimate and heroku

[WIP #27]

chore(configure): configure codeclimate and heroku

configure codeclimate and heroku
added CI/CD for codeclimate and heroku

[WIP #27]

chore(configure): configure codeclimate and heroku

configure codeclimate and heroku
added CI/CD for codeclimate and heroku

[WIP #27]

PrinceNiyonshuti committed:
* chore(setup): setup redux

setup redux
setup actions
setup reducers
setup store
setup constants

 [ Finishes #26 ]

* Fix(Setup): setup redux

fixed the ReactDom issue that was causing the DOM to misbehave
fixed the webpack issue in index.html that was prevent the DOM to be manipulated

[ Finishes #26 ]

* Fix (Conflict) : Conflict resolution

fixing the conflict resolution on this branch

Niyonkuru1 committed:
feat(create UI landing): create UI of the landing page

creating components of the landing: NavBar, LandiPage and FooterBlock components
adding assets folder and create images folder inside them
including the tailwind css using CDN
trying to configure Tailwind locally using packages but currently THEY ARE NOT WORKING

[ WIP #28 ]

Niyonkuru1 committed:
feat(create UI landing): create UI of the landing page

- Uncommenting the CDN import statement in the index.html

[ WIP #28 ]

Niyonkuru1 committed:
feat(create UI landing): create UI of the landing page

- Uncommenting the CDN import statement in the index.html

[ WIP #28 ]

Niyonkuru1 committed:
feat(create UI landing): create UI of
[ WIP #28 ]

Niyonkuru1 committed:
feat(create UI landing): create UI of the landing page

- separating the components.

[ WIP #28 ]

PrinceNiyonshuti reviewed (approved):
Looks good

Niyonkuru1 committed:
trying to install the tailwind

Commit:
Initial tailwindcss setup

Niyonkuru1 committed:
chore(setup): setup tailwind

- installing minisccextractor packages
- configuring the webpack to use it with tailwind
[ Finishes #3 ]

Niyonkuru1 committed:
chore(setup): setup tailwind

- cleaning the code
[ Finishes #3 ]

thepatrickniyo commented on packageDraft.txt:
```diff
@@ -0,0 +1,50 @@
+// {
+//   "name": "setup-react-project",
+//   "version": "1.0.0",
+//   "description": "setting up the react project for phantom app",
+//   "main": "index.js",
+//   "scripts": {
+//     "start": "webpack serve --mode development --open",
+//     "build": "webpack --mode production",
+//     "test": "react-scripts test --watchAll=false",
+//     "coverage": "react-scripts test --coverage --watchAll=false"
+//   },
+//   "keywords": [
+//     "webpack5",
+//     "babel",
+//     "react"
+//   ],
+//   "author": "elite team member",
+//   "license": "ISC",
+//   "dependencies": {
+//     "babel-jest": "^27.5.1",
+//     "dotenv": "^16.0.0",
+//     "dotenv-webpack": "^7.1.0",
+//     "jest": "^27.5.1",
+//     "path": "^0.12.7",
+//     "react": "^18.0.0",
+//     "react-dom": "^18.0.0",
+//     "react-redux": "^7.2.8",
+//     "react-router": "^6.3.0",
+//     "react-router-dom": "^6.3.0",
+//     "react-scripts": "^5.0.0",
+//     "react-test-renderer": "^18.0.0",
```
Can you please try to remove the comments.

thepatrickniyo reviewed (changes requested):
Remove the commented source codes then we be able to merge the work

Niyonkuru1 committed:
feat(landing-page): implement responsiveness

- adding responsiveness to the page using tailwind media queries
- adding footer background image
- adding custom font-family, colors,  through tailwind.config.js
- removing and refactoring commented codes for clean up.
-

[ WIP #28 ]

Niyonkuru1 committed:
feat(landing-page): finalising responsiveness and adding validation to the subscribe form and increasing test coverage.

- installing formik and use it to validate the form for subscribing to the site
- resolving the issues of rendering the full screen on the tablet mode
- cleaning the code
- adding the color text of errors rendering configulation in the tailwind.config.js
- adding the images to the middle of the page
- increasing test coverage using testing react library

[ Finishes #28 ]

PrinceNiyonshuti reviewed (approved):
looks good

thepatrickniyo commented:
Remove the conflicts in this branch

Niyonkuru1 committed:
feat(landing-page): resolving merge conflicts.

- Resolving conflicts with merging

[ Finishes #28 ]

Niyonkuru1 committed:
feat(landing-page): resolving merge conflicts - second.

- Resolving conflicts with merging 2.

[ Finishes #28 ]

Niyonkuru1 committed:
feat(landing-page): adding correct fonts to the page and shadow on the image.

- Adding correct font as depicted in mockup and putting the shadow on the image

[ Finishes #28 ]
